src/InputProcessor.ts

```diff
@@ -0,0 +1,28 @@
+
+import { AbstractInputProcessor } from "models/AbstractInputProcessor";
+import { CommonInputModel } from "models/CommonInputModel";
+import { AsyncAPIInputProcessor } from "processors/AsyncAPIInputProcessor";
+import { JsonSchemaInputProcessor } from "processors/JsonSchemaInputProcessor";
+
+/**
+ * Main input processor which figures out the type of input it receives and delegates the processing into separate individual processors.
+ */
+export class InputProcessor extends AbstractInputProcessor {
+  asyncAPIProcessor: AbstractInputProcessor;
```

Comment:
I think that we should have more flexible this class. I mean at the moment we have only processors for AsyncAPI and JSON Schema. Better idea to store processors than fields will be container as Map:

```ts
export class InputProcessor extends AbstractInputProcessor {
  processors: Map<string, AbstractInputProcessor>;
  
  constructor() { super(); }
  
  addProcessor(type: string, processor: AbstractInputProcessor) {
    this.processors.set(type, processor);
  }
  
  async process(object: any, type: string = 'json-schema'): Promise<CommonInputModel> {
    const processor = this.processors.get(type);
    if (processor === undefined) {
      throw Error(...)
    }
    try {
      return processor. process(object);
    } catch(err) {
       ... err
    }
  }
}

const mainProcessor = new InputProcessor();
mainProcessor.addProcessor('json-schema', new JsonSchemaInputProcessor());
mainProcessor.addProcessor('asyncapi', new AsyncAPIInputProcessor());
```

What do you think? I know that I changed the signature of `process` function but I have a problem what we want to handle another inputs that these two (of course in future).